lecture_4/04_event_settler/api/eventSettler.go

```diff
@@ -0,0 +1,77 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"04_event_settler/models"
+)
+
+const ipAddress = "http://127.0.0.1"
+const eventApiPort = ":8080"
+const eventPath = "/event/update"
+
+// Arrays can't be constants https://stackoverflow.com/questions/13137463/declare-a-constant-array
+var outcomes = [...]string{"won", "lost"}
+
+func GetOddsWithStatus(client http.Client, url string) ([]models.BetDto, error) {
+	response, err := client.Get(url)
+	defer response.Body.Close()
+	if err != nil {
+		return []models.BetDto{}, errors.Wrap(err, "error getting data from bets API")
+	}
+
+	bodyContent, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return []models.BetDto{}, errors.Wrap(err, "error reading response body")
+	}
+
+	var decodedBets []models.BetDto
+	err = json.Unmarshal(bodyContent, &decodedBets)
+	if err != nil {
+		return []models.BetDto{}, errors.Wrap(err, "error unmarshalling response body")
+	}
+
+	return decodedBets, nil
+}
+
+func ResolveBets(client http.Client, bets []models.BetDto) error {
+	// UnixNano() is recommended by https://golang.org/pkg/math/rand/
+	// example: https://play.golang.org/p/8qOg760mLNi
+	rand.Seed(time.Now().UnixNano())
+
+	// set is used for storing all events from all bets, some bets have multiple events
+	setForEvents := map[string]bool{}
```

Comment:
This is fine as an implementation. You can also use an empty struct as the value in the map to represent sets. Empty structs work the same as bools, but take up less space. 
You don't have to change this, just fyi.

Reply:
I first wrote it as `map[string]struct{}`, I know that those are 0 byte `structs`, but when i checked how others did it, they mostly used `bool`, and the code looked cleaner, giving it value of `true` instead of `struct{}{}`.